### PR TITLE
cli: refactor kubernetes package

### PR DESCRIPTION
**Pull request by daniel-weisse**
### Context
The CLI Kubernetes package needs some clean up.
The goal of this PR is:
* Make it clearer what the package does: It is supposed to be used for highlevel interactions with Kubernetes resources by the CLI
  * Low level stuff, like creating a resource or updating a resource (single function calls), should be kept in `internal/kubectl`
* Clean up the logic of some commands
* Remove unused functions

### Proposed change(s)

- Rename the `kubernetes` package to `kubecmd` to better highlight its intended use
- Rename `Upgrader` to `KubeCmd`
- Remove exported functions which take a client to interact with the kubernetes API as argument
  - Replaced with methods of the `kubecmd` client
- Replace usage of custom unstructured and structured Kubernetes API client with the client from `internal/kubernetes`
- Rename some variables and functions to better highlighted their functionality and intended use case

### CODE REVIEW

**Review by 3u13r — APPROVED** [review]
LGTM

**Review by malt3 — APPROVED** [review]

File: cli/internal/cmd/upgradeapply.go [hunk 1]
```diff
@@ -369,10 +370,7 @@ func (u *upgradeApplyCmd) confirmIfUpgradeAttestConfigHasDiff(cmd *cobra.Command
 			return errors.New("aborting upgrade since attestation config is different")
 		}
 	}
-	// TODO(elchead): move this outside this function to remove the side effect.
-	if err := u.kubeUpgrader.BackupConfigMap(cmd.Context(), constants.JoinConfigMap); err != nil {
-		return fmt.Errorf("backing up join-config: %w", err)
-	}
+
```
Comment (derpsteb) [inline cli/internal/cmd/upgradeapply.go:373]: Is this done somewhere else now?

File: cli/internal/kubecmd/kubecmd.go
Comment (derpsteb) [inline cli/internal/kubecmd/kubecmd.go:1]: maybe add a pkg description?

File: cli/internal/cmd/upgradeapply.go (hunk 1 above)
Comment (daniel-weisse) [reply cli/internal/cmd/upgradeapply.go:373]: Yes, see: https://github.com/edgelesssys/constellation/blob/14b470192fc09f1f101f8c1fac79b7b92a7ae2ba/cli/internal/kubecmd/kubecmd.go#L186-L216

File: cli/internal/kubecmd/kubecmd.go
Comment (derpsteb) [inline cli/internal/kubecmd/kubecmd.go:1]: maybe add a pkg description?

**Review by derpsteb — APPROVED** [review]

**Comment by daniel-weisse** [comment]
e2e tests:

https://github.com/edgelesssys/constellation/actions/runs/5926178789
https://github.com/edgelesssys/constellation/actions/runs/5926181155
https://github.com/edgelesssys/constellation/actions/runs/5926183566

**Pull request merged**